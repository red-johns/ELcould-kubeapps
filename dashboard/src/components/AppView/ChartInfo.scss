.ChartInfo > a {
  display: block;
  width: 100%;
}
