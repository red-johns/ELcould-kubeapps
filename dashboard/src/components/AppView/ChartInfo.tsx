import * as React from "react";

import { hapi } from "shared/hapi/release";
import Card, { CardContent, CardFooter, CardGrid, CardIcon } from "../Card";

import placeholder from "../../placeholder.png";
import "./ChartInfo.css";

interface IChartInfoProps {
  app: hapi.release.Release;
}

class ChartInfo extends React.Component<IChartInfoProps> {
  public render() {
    const { app } = this.props;
    const name = app.name;
    const metadata = app.chart && app.chart.metadata;
    const icon = metadata && metadata.icon;
    const iconSrc = icon ? icon : placeholder;
    let notes = <span />;
    if (metadata) {
      notes = (
        <div>
          {metadata.appVersion && <div>App Version: {metadata.appVersion}</div>}
          <div>Chart Version: {metadata.version}</div>
        </div>
      );
    }
    return (
      <CardGrid className="ChartInfo">
        <Card>
          <CardIcon icon={iconSrc} />
          <CardContent>
            <h5>{name}</h5>
            <p className="margin-b-reset">{metadata && metadata.description}</p>
          </CardContent>
          <CardFooter>
            <small>{notes}</small>
          </CardFooter>
        </Card>
      </CardGrid>
    );
  }
}

export default ChartInfo;
