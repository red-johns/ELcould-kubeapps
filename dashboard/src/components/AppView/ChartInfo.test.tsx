import { shallow } from "enzyme";
import * as React from "react";

import { hapi } from "shared/hapi/release";
import ChartInfo from "./ChartInfo";

it("renders a app item", () => {
  const wrapper = shallow(
    <ChartInfo
      app={
        {
          chart: {
            metadata: {
              appVersion: "0.0.1",
              description: "test chart",
              icon: "icon.png",
              version: "1.0.0",
            },
          },
          name: "foo",
        } as hapi.release.Release
      }
    />,
  );
  expect(wrapper.find(".ChartInfo").exists()).toBe(true);
  expect(wrapper).toMatchSnapshot();
});
