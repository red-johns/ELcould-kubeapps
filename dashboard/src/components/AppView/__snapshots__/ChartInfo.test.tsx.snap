// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a app item 1`] = `
<CardGrid
  className="ChartInfo"
>
  <Card>
    <CardIcon
      icon="icon.png"
    />
    <CardContent>
      <h5>
        foo
      </h5>
      <p
        className="margin-b-reset"
      >
        test chart
      </p>
    </CardContent>
    <CardFooter>
      <small>
        <div>
          <div>
            App Version: 
            0.0.1
          </div>
          <div>
            Chart Version: 
            1.0.0
          </div>
        </div>
      </small>
    </CardFooter>
  </Card>
</CardGrid>
`;
